proxmoxtf/resource_virtual_environment_vm.go

```diff
@@ -76,6 +76,7 @@ const (
 	dvResourceVirtualEnvironmentVMInitializationUserAccountPassword = ""
 	dvResourceVirtualEnvironmentVMInitializationUserDataFileID      = ""
 	dvResourceVirtualEnvironmentVMInitializationVendorDataFileID    = ""
+	dvResourceVirtualEnvironmentVMInitializationNetworkConfigFileID = ""
```

Comment:
Could we change "NetworkConfigFile" -> "NetworkDataFile" here and below, and argument `network_config_file_id` -> `network_data_file_id` as well?
I know it looks a bit out of context this way, but I'd like to keep consistency in naming with other cloud-init parts (at the lines above)

Reply:
Updated the code so it's using the network data naming!

I've tested the modification on my proxmox cluster and it worked as expected. I just provided only `user_data_file_id` and `network_data_file_id` parameters (as in the example) and it created the cloud-init drive with both of these files. Combining the network-config file with `ip_config` or dns related configuration in the `initialization` block is probably not going to work. (but also wouldn't really make sense)

Because we are not specifying any network related configuration on the PVE  level, it isn't creating the network-config file, and we are free to provide it ourselves. The build in generation system still uses cloud-init network config v1, but I've been testing this with Ubuntu and thus provided my config in the v2 syntax. The nice thing is that we are no longer limited to what PVE supports as it's no longer managing the file in this case.

I did just notice that second runs are still a bit misbehaving and it wants to re-add the `network_data_file_id` attribute again. So I might have forgotten something to prevent that behavior. (ignoring it does seem to give me a functioning system, so probably easy to solve)

While testing I also noticed that the tag ordering is also causing issues similar to #185, because pve is expecting them in alphabetical order.

Reply:
Awesome! I also found an error in my netplan config that caused issues on apply in my tests, so everything is working now.
I've added the new argument to the docs and also fixed the re-apply use case, so I think the PR is good to go.

I'll open a separate ticket for tags reordering, this bugs me too...

Reply:
Great, thank you!